src/java.desktop/macosx/classes/apple/laf/JRSUIControl.java

```diff
@@ -109,17 +113,26 @@ public JRSUIControl(final boolean flipped){
         flipped = other.flipped;
         cfDictionaryPtr = getCFDictionary(flipped);
         if (cfDictionaryPtr == 0) throw new RuntimeException("Unable to create native representation");
+        Disposer.addRecord(disposerReferent, new JRSUIControlDisposerRecord(cfDictionaryPtr));
         nativeMap = new HashMap<Key, DoubleValue>();
         changes = new HashMap<Key, DoubleValue>(other.nativeMap);
         changes.putAll(other.changes);
     }
 
-    @Override
-    @SuppressWarnings("removal")
-    protected synchronized void finalize() throws Throwable {
-        if (cfDictionaryPtr == 0) return;
-        disposeCFDictionary(cfDictionaryPtr);
-        cfDictionaryPtr = 0;
+    private static class JRSUIControlDisposerRecord implements DisposerRecord {
+
+        private long cfDictionaryPtr;
+        JRSUIControlDisposerRecord(long ptr) {
+            cfDictionaryPtr = ptr;
+        }
+
+        public void dispose() {
+            try {
+                disposeCFDictionary(cfDictionaryPtr);
+            } catch (Throwable t) {
+            }
+            cfDictionaryPtr = 0;
```

Comment:
cfDictionaryPtr = 0 seems is not needed as well we do not reset pointers in other classes -> then we can mark cfDictionaryPtr as final.

Reply:
it can already be made final in the main class but I can do it as well in the DisposerRecord